### ISSUE TITLE
fix adding data for different geometries

### ISSUE THREAD

**Issue by mulvrova** [StagPython/StagPy#64]
when using spherical annulus geometry, data are added at the end of the fields to make the plots continuous at phi=0=2pi. this is not necessary for other geometries such as cartesian.

**Comment by amorison** [comment]
This isn't as easy a fix as it might look.

The reason an extra ghost point is added to fields at such a low level isn't purely cosmetic. It is there to ensure the rest of the code will receive fields (vector or not), coordinates (`geom.*_coord`), and meshes (`geom.*_mesh`) with consistent shapes. This had repercussions in the `field` and `plates` module where the assumption these variables have consistently an extra point is hard coded. The `plates` module can't be tested as long as #34 isn't done, so this has to be done before attempting to address this issue. More thorough testing of the `field` module also has to be implemented to be sure not to break anything.

---

- Addition of an extra point is implemented in `_step._Fields.__getitem__` and `_step._Geometry.__init__`.
- Ideally, one would want to add that extra point to fields without ghost points only when plotting a full spherical annulus (i.e. with an aspect ratio above 2pi) or a full cylindrical annulus if that geometry still exists.
- The values in the layer to add depend on the boundary condition, it should be an average of the first and last columns if the walls are wrapped (`(x|y)_bc='ww'`, this is what's always done for now), something else if the BC is different.
- A possible implementation would be to code the addition of the layer as a function in `field` that would be called by `plot_scalar` (and others where appropriate). I gave a quick stab at it (only for spherical annulus, and the same should be applied to `xmesh` and `ymesh`):
```python
def _fix_shape(step, fld):
    """Add extra ghost point to field if necessary."""
    if not (step.geom.spherical and step.geom.twod_yz and
            step.sdat.par['geometry']['aspect_ratio'] >= 2 * np.pi):
        return fld
    if fld.shape[0] == step.geom.nytot:
        # add extra ghost point
        if step.sdat.par['boundaries']['y_bc'] == 'ww':
            newline = (fld[:1] + fld[-1:]) / 2
        else:
            # depends on the boundary condition...
            newline = fld[-1]
        fld = np.concatenate((fld, newline), axis=0)
```

Such an approach would rid us of unnecessary extra layers. On the other hand, one would have to be careful that vector fields (and maybe some scalar fields such as pressure with the old output format of StagYY) would no longer have a shape that is consistent with `geom.*_coord` and `geom.*_mesh` since they have an extra ghost point. This will require some care in the associated functions in the `field` and `plates` modules.

**Comment by amorison** [comment]
This is solved by PR #73. A newline is only added when plotting a scalar field in 2D annulus geometry if smoothing is requested via the `field.interpolate` config variable (or the `+interpolate` CLI switch). Otherwise, we use the fact that `pcolormesh` can be given the wall coordinates instead of the center coordinates and therefore have a full annulus without any tweaking.
